public/app/features/query/state/PanelQueryRunner.ts

```diff
@@ -451,6 +470,31 @@ export class PanelQueryRunner {
   getLastRequest(): DataQueryRequest | undefined {
     return this.lastRequest;
   }
+
+  shouldAddErrorWhenDatasourceVariableIsMultiple(
+    datasource: DataSourceRef | DataSourceApi | null,
+    scopedVars: ScopedVars | undefined
+  ): boolean {
+    let addWarningMessageMultipleDatasourceVariable = false;
+
+    if (datasource?.uid?.startsWith('${')) {
+      const variableName = this.templateSrv.getVariableName(datasource.uid) ?? '';
+      const variable = this.templateSrv.getVariables().find((v) => v.name === variableName);
```

Comment:
Instead of this lookup I figured out a cleaner / simpler approach to get the array of values (if it is multi ) , a solution that works in both old and scenes 

By calling templateSrv.replace with a custom format function you get the raw value , checking it this is an array is enough here 


https://github.com/grafana/grafana/pull/83356/files

Reply:
Hey @torkelo 👋🏾 , I added the suggestion, however, the unit test is not great, I had to test implementation details, because of the TemplateSrvMock we are using ([change](https://github.com/grafana/grafana/pull/76039/commits/f483949216dd7728c391d9c85552e1d825285218#diff-bd32a0cd9404bec9133154b57c384650f134a84848d22f25f7748c9a71d92452L465))